src/main/java/org/openrewrite/maven/RewriteDiscoverMojo.java

```diff
@@ -145,4 +149,50 @@ private void writeRecipeDescriptor(RecipeDescriptor rd, boolean verbose, int cur
             }
         }
     }
+    
+    private void promptToRun(RecipeDescriptor rd) throws MojoExecutionException {
+        try {
+            String prompted = prompter.prompt(String.format(RUN_PROMPT_MESSAGE, rd.getDisplayName()), Arrays.asList("y", "n"), "n");
+            if ("y".equalsIgnoreCase(prompted)) {
+
+                RewriteRunMojo rewriteRunMojo = new RewriteRunMojo();
+                rewriteRunMojo.rewriteActiveRecipes = rd.getName();
+
+                rewriteRunMojo.project = project;
+                rewriteRunMojo.runtime = runtime;
+                rewriteRunMojo.settingsDecrypter = settingsDecrypter;
+                rewriteRunMojo.repositorySystem = repositorySystem;
+                rewriteRunMojo.mavenSession = mavenSession;
+                
+                rewriteRunMojo.configLocation = configLocation;
+                rewriteRunMojo.activeRecipes = activeRecipes;
+                rewriteRunMojo.activeStyles = activeStyles;
+                rewriteRunMojo.rewriteActiveStyles = rewriteActiveStyles;
+                rewriteRunMojo.metricsUri = metricsUri;
+                rewriteRunMojo.metricsUsername = metricsUsername;
+                rewriteRunMojo.metricsPassword = metricsPassword;
+                rewriteRunMojo.pomCacheEnabled = pomCacheEnabled;
+                rewriteRunMojo.pomCacheDirectory = pomCacheDirectory;
+                rewriteRunMojo.skipMavenParsing = skipMavenParsing;
+                rewriteRunMojo.checkstyleConfigFile = checkstyleConfigFile;
+                rewriteRunMojo.sizeThresholdMb = sizeThresholdMb;
+                rewriteRunMojo.failOnInvalidActiveRecipes = failOnInvalidActiveRecipes;
+                rewriteRunMojo.runPerSubmodule = runPerSubmodule;
+
+                // Non visible fields
+//                rewriteRunMojo.exclusions = exclusions;
+//                rewriteRunMojo.rewriteExclusions = rewriteExclusions;
+//                rewriteRunMojo.plainTextMasks = plainTextMasks;
+//                rewriteRunMojo.rewritePlainTextMasks = rewritePlainTextMasks;
+//                rewriteRunMojo.recipeArtifactCoordinates = recipeArtifactCoordinates;
+//                rewriteRunMojo.computedRecipes = computedRecipes;
+//                rewriteRunMojo.computedStyles = computedStyles;
+//                rewriteRunMojo.computedRecipeArtifactCoordinates = computedRecipeArtifactCoordinates;
```

Comment:
```suggestion
```
These fields I could not override, as they are not public; Don't think there's any issue with leaving these out.